axelrod/strategies/axelrod_second.py

```diff
@@ -566,3 +566,156 @@ def strategy(self, opponent: Player) -> Action:
+       cooperate for a single turn.
+    2. If in the last three turns, the player/opponent acted differently from
+       each other and they're alternating, then change next defect to
+       cooperate.  (Doesn't block third rule.)
+    3. Otherwise, do tit-for-tat.
+
+    Start in normal mode, but every 25 turns starting with the 27th turn, 
+    re-evaluate the mode.  Enter defect mode if any of the following 
+    conditions hold:
+
+    - Detected random:  Opponent cooperated 7-18 times since last mode 
+      evaluation (or start) AND less than 70% of opponent cooperation was in
+      response to player's cooperation, i.e.
+      cc_count / (cc_count+cd_count) < 0.7
+    - Detect defective:  Opponent cooperated fewer than 3 times since last mode
+      evaluation.
+
+    When switching to defect mode, defect immediately.  The first two rules for
+    normal mode require that last three turns were in normal mode.  When starting
+    normal mode from defect mode, defect on first move.
+
+    Names:
+
+    - Borufsen: [Axelrod1980b]_
+    """
+
+    name = "Borufsen"
+    classifier = {
+        'memory_depth': float('inf'),
+        'stochastic': False,
+        'makes_use_of': set(),
+        'long_run_time': False,
+        'inspects_source': False,
+        'manipulates_source': False,
+        'manipulates_state': False
+    }
+
+    def __init__(self):
+        super().__init__()
+        self.cd_counts, self.cc_counts = 0, 0
+        self.mutual_defect_streak = 0
+        self.echo_streak = 0
+        self.flip_next_defect = False
+        self.mode = "Normal"
+
+    def try_return(self, to_return):
+        """
+        We put the logic here to check for the `flip_next_defect` bit here,
+        and proceed like normal otherwise.
+        """
+
+        if to_return == C:
+            return C
+        # Otherwise look for flip bit.
+        if self.flip_next_defect:
+            self.flip_next_defect = False
+            return C
+        return D
+
+    def strategy(self, opponent: Player) -> Action:
+        turn = len(self.history) + 1
+
+        # Nip this case in the bud.
+        if turn == 1:
+            return C
+
+        # Update the response history.
+        if turn >= 3:
+            if opponent.history[-1] == C:
+                if self.history[-2] == C:
+                    self.cc_counts += 1
+                else:
+                    self.cd_counts += 1
+
+        # Check if it's time for a mode change.
+        if turn > 2 and turn%25 == 2:
+            coming_from_defect = False
+            if self.mode == "Defect":
+                coming_from_defect = True
+
+            self.mode = "Normal"
+            coops = self.cd_counts + self.cc_counts
+
+            # Check for a defective strategy
+            if coops < 3:
+                self.mode = "Defect"
+
+            # Check for a random strategy
+            if (coops>=8 and coops<=17) and self.cc_counts*1.0/coops < 0.7:
```

Comment:
PEP8:

```
(coops >= 8 and coops <= 17) and self.cc_counts / coops < 0.7:
```

(I believe there's no need for `1.0 / coops` as this is py3+ so will assume float division).